wordpress-mobile/WordPress-Android
# [Issue/20295]  Location metadata in uploaded pictures

Fixes #20295

-----

This is the companion PR for https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/3030 

Merge Instructions

- Make sure the FLuxC PR has been merged to trunk
- Grab the FluxC trunk has from CI and update build.gradle
- Merge as normal

-----

This PR aims to resolve the issues described above. After investigating, I found that there were two issues:

1. When the "Optimize Images" option was enabled, we didn't copy the ExifInterface data (which includes location metadata) to the new optimized file.

2. When the "Optimize Images" option was disabled, we could not access the location data in the latest Android API versions because we were not requesting the ACCESS_MEDIA_LOCATION permission.

This PR requests the necessary permissions and uses the new ExifUtils functions to copy the ExifInterface data to the optimized media files.

## To Test:
**First check the existing problem:** 

1. Install on your device the latest version of Jetpack app from PlayStore.
2. Log in. 
3. Tap on Me -> App Settings -> Verify that "Optimize Images" is enabled and "Remove location from media" is disabled.
4. Go to My Site -> Media -> Tap on the "+" icon at the top right corner -> Choose from device 
5. Give all the permissions asked. 
6. Choose an image from your device and upload
7. Find the uploaded image in the media library and open it
8. Tap the "share" icon at the top right corner
9. Choose your email client like Gmail and send it to your email address
10. Open the mail from your browser, tap on the link of the image. 
11. Right click on the image and save it on your computer. 
12. Open the image using "Preview" on your Mac, and click on Tools -> Show Inspector. 
13. Tap on the "More Info" icon
![Screen Shot 2024-06-07 at 18 38 50 PM](https://github.com/wordpress-mobile/WordPress-Android/assets/5176760/e533e2d3-3446-47c6-b987-82b5f17627df)

- [x] Verify that there are no Exif or GPS data. 

14. Tap on Me -> App Settings -> Disable "Optimize Images". 
15. Repeat steps 4-13 
- [x] Verify that there are Exif data and GPS are zero. 

**Test Solution**
16. Install the app from this PR.
17. Repeat steps 4-13 
- [x] Verify that Exif and GPS data exist
18. Repeat steps 14-15
- [x] Verify that Exif and GPS data exist
19. Try to upload an image from the editor while creating a post using an image block.
- [x] Verify that Exif and GPS data exist

ATTENTION: If users chooses to "Take a photo" from our app and not choose a photo from the device, no location data are saved in the uploaded image. This functionality in not in the scope of this PR. 



-----

## Regression Notes

1. Potential unintended areas of impact

    - N/A

14. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual testing

-----
